### PR TITLE
Update usage of useDeleteItems

### PR DESCRIPTION
**Pull request by nixocio**
* Update usage of useDeleteItems.
* Remove async when not necessary
* Update return of promises

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

### CODE REVIEW

**Comment by nixocio** [comment]
Currently running e2e tests.

**Comment by unlikelyzero** [comment]
@nixocio if the tests pass, please merge! Right now we have a bug with the yolok pipeline: if a visual test fails, it will impact the results on the functional tests. We have a ticket to fix that https://github.com/ansible/tower-qa/issues/5711

For now, just run this PR without the visual test option

File: awx/ui_next/src/screens/Application/ApplicationDetails/ApplicationDetails.jsx [hunk 1]
```diff
@@ -21,11 +21,11 @@ function ApplicationDetails({
...
     isLoading: deleteLoading,
-    deletionError,
-    deleteItems: handleDeleteApplications,
+    error: deletionError,
+    request: deleteApplications,
     clearDeletionError,
-  } = useDeleteItems(
-    useCallback(async () => {
+  } = useCallback(
+    useRequest(async () => {
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Application/ApplicationDetails/ApplicationDetails.jsx]: This should be reversed to `useRequest(useCallback(...`

File: awx/ui_next/src/screens/Application/ApplicationDetails/ApplicationDetails.jsx [hunk 2]
```diff
@@ -21,11 +21,11 @@ function ApplicationDetails({
   const history = useHistory();
   const {
     isLoading: deleteLoading,
-    deletionError,
-    deleteItems: handleDeleteApplications,
+    error: deletionError,
+    request: deleteApplications,
     clearDeletionError,
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Application/ApplicationDetails/ApplicationDetails.jsx]: useRequest doesn't return a `clearDeletionError`. You'll have to pass `deletionError` into a `useDismissableError` below this to get the clearDeletionError function

**Review by keithjgrant — CHANGES_REQUESTED** [review]
Looks good for the most part. Just a couple issues in one file

File: awx/ui_next/src/screens/Application/ApplicationDetails/ApplicationDetails.jsx (hunk 1 above)
Comment (nixocio) [reply awx/ui_next/src/screens/Application/ApplicationDetails/ApplicationDetails.jsx]: @keithjgrant, updates per comments.

**Review by keithjgrant — APPROVED** [review]
lgtm!

**Comment by nixocio** [comment]
I ran the visual tests and e2e tests, no regression was added by those changes.

**Pull request merged**